lib/src/global_packages.dart

```diff
@@ -378,6 +378,22 @@ To recompile executables, first run `global decativate ${dep.name}`.
       }
     });
 
+    // Check that the SDK constraints the lockFile says we have are honored.
+    lockFile.sdkConstraints.forEach((sdkName, constraint) {
+      var sdk = sdks[sdkName];
+      if (sdk == null) {
+        dataError('${log.bold(name)} as globally activated requires '
+            'unknown SDK "$name".');
+      } else if (sdkName == 'dart') {
+        if (constraint.allows(sdk.version)) return;
+        dataError("${log.bold(name)} as globally activated doesn't "
+            'support Dart ${sdk.version}, try to re-activate it.');
```

Comment:
Could we give a command they could copy-paste to do that?
```
pub global activate <name> --source <source>
```
It probably requires some logic to handle  reproduction of the `source` and `name`...

I suspect they would not need to deactivate first...

Reply:
I omitted the `--source` part... I see no reason we shouldn't do it. I just wanted to close this now :D